File: backend/app/schemas/progress.py
from pydantic import BaseModel


class ModuleProgressItem(BaseModel):
    id: int
    title: str
    pct: float
    completed: int
    total: int


class ProgressResponse(BaseModel):
    overall_pct: float
    total_time_seconds: int
    topics_completed: int
    quiz_avg_score: float | None = None
    modules: list[ModuleProgressItem] = []


class StreakResponse(BaseModel):
    current_streak: int
    longest_streak: int
    last_active_date: str | None = None

File: frontend/src/types/progress.ts
export interface ModuleProgressItem {
  id: number
  title: string
  pct: number
  completed: number
  total: number
}

export interface ProgressData {
  overall_pct: number
  total_time_seconds: number
  topics_completed: number
  quiz_avg_score: number | null
  modules: ModuleProgressItem[]
}

export interface ActivityItem {
  type: string
  description: string
  timestamp: string
}

export interface StreakResponse {
  current_streak: number
  longest_streak: number
  last_active_date: string | null
}
